typing/typedecl.ml

```diff
@@ -179,15 +180,20 @@ let set_fixed_row env loc p decl =
         let row = Btype.row_repr row in
         Btype.set_type_desc tm
           (Tvariant {row with row_fixed = Some Fixed_private});
-        if Btype.static_row row then Btype.newgenty Tnil
+        if Btype.static_row row then
+          (* the syntax hinted at the existence of a row variable,
+             but there is in fact no row variable to make private, e.g.
+             [ type t = private [< `A > `A] ] *)
```

Comment:
The comment here indicates that we know something about the user intent ("the syntax"), but the comment on the function declaration just says that "is_fixed_type" is a precondition. Maybe the comment on the function declaration could also, for people who are not expert on what "is_fixed_type" means, point out that we are working with a private type declaration?

Reply:
Good point. I would propose to update the name of the function to `set_private_row` too.

Reply:
why not, but many places use `fixed` as an adjective, should they be updated too?

Reply:
A private row is a subcase of fixed rows, but `Typedecl` and `set_fixed_row` only generate this `Fixed_private` subcase.